source/Calamari/Kubernetes/ResourceStatus/ResourceStatusChecker.cs

```diff
@@ -30,75 +41,144 @@ public class ResourceStatusChecker : IResourceStatusChecker
         internal const string MessageDeploymentSucceeded = "Resource status check completed successfully because all resources are deployed successfully";
         internal const string MessageDeploymentFailed = "Resource status check terminated with errors because some resources have failed";
         internal const string MessageInProgressAtTheEndOfTimeout = "Resource status check terminated because the timeout has been reached but some resources are still in progress";
-        
+
         private readonly IResourceRetriever resourceRetriever;
         private readonly IResourceUpdateReporter reporter;
         private readonly ILog log;
+        private ITimer timer;
+        private readonly HashSet<ResourceIdentifier> resources = new HashSet<ResourceIdentifier>();
```

Comment:
`HashSet<>` is used to enforce uniqueness. I've also added equality operators to ResourceIdentifier because we can't use `record`s here. 😞